cmd/derper/derper.go

```diff
@@ -460,3 +464,21 @@ func (l *rateLimitedListener) Accept() (net.Conn, error) {
 	l.numAccepts.Add(1)
 	return cn, nil
 }
+
+// logFilter is used to filter out useless error logs that are logged to
+// the net/http.Server.ErrorLog logger.
+type logFilter struct{}
+
+func (*logFilter) Write(p []byte) (int, error) {
+	if bytes.HasSuffix(p, []byte(": EOF\n")) ||
+		bytes.HasSuffix(p, []byte(": i/o timeout\n")) ||
+		bytes.HasSuffix(p, []byte(": read: connection reset by peer\n")) ||
+		bytes.HasSuffix(p, []byte(": remote error: tls: bad certificate\n")) ||
+		bytes.HasSuffix(p, []byte(": tls: first record does not look like a TLS handshake\n")) {
```

Comment:
I forget if this ever got faster. It used to suck with those byte slices.

I still kinda prefer mem.HasSuffix(mem.B(..), mem.S(...))

Reply:
Tested this out in godbolt (https://godbolt.org/z/1EE7dzbKz) and verified that with byte slices, it constructs a new one on each call to this function, whereas with `mem.S` they're static. Switched to that.